<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@sigmamessaging/otp-sdk](./otp-sdk.md) &gt; [SigmaOtpSDK](./otp-sdk.sigmaotpsdk.md) &gt; [waitForChannelSelected](./otp-sdk.sigmaotpsdk.waitforchannelselected.md)

## SigmaOtpSDK.waitForChannelSelected() method

Возвращает текущий канал после смены статуса на [EOtpHandlerChannelStatus.sent](./otp-sdk.eotphandlerchannelstatus.md)

**Signature:**

```typescript
waitForChannelSelected(requestId: string): Promise<IOtpChannelData>;
```

## Parameters

<table><thead><tr><th>

Parameter


</th><th>

Type


</th><th>

Description


</th></tr></thead>
<tbody><tr><td>

requestId


</td><td>

string


</td><td>

ID процесса


</td></tr>
</tbody></table>

**Returns:**

Promise&lt;[IOtpChannelData](./otp-sdk.iotpchanneldata.md)<!-- -->&gt;

[IOtpChannelData](./otp-sdk.iotpchanneldata.md)

## Exceptions

[Exceptions.ResendChannelException](./otp-sdk.exceptions.resendchannelexception.md)

[Exceptions.SessionNotFoundException](./otp-sdk.exceptions.sessionnotfoundexception.md)

[Exceptions.AttemptsExpiredException](./otp-sdk.exceptions.attemptsexpiredexception.md)

[Exceptions.NoAvailableChannelsException](./otp-sdk.exceptions.noavailablechannelsexception.md)

[Exceptions.SessionClosedException](./otp-sdk.exceptions.sessionclosedexception.md)

[Exceptions.LongPollingTimeoutException](./otp-sdk.exceptions.longpollingtimeoutexception.md)

[Exceptions.NoConnectionException](./otp-sdk.exceptions.noconnectionexception.md)

[Exceptions.InternalServerErrorException](./otp-sdk.exceptions.internalservererrorexception.md)

[Exceptions.CommonForbiddenException](./otp-sdk.exceptions.commonforbiddenexception.md)

## Remarks

Это LongPolling вариант запроса [SigmaOtpSDK.getChannel()](./otp-sdk.sigmaotpsdk.getchannel.md) и отличается тем, что ждёт пока какая либа информация о канале не обновится. Возвращает ошибку [Exceptions.LongPollingTimeoutException](./otp-sdk.exceptions.longpollingtimeoutexception.md) при которой надо выполнить запрос заново.

